browser/src/docstatefunctions.js

```diff
@@ -335,3 +335,23 @@ app.impress.isSlideSelected = function (index) {
 		return app.impress.partList[index].selected === 1;
 	} else return false;
 };
+
+app._flushLayoutingQueue = function () {
+	if (app.layoutTasks.length) {
+		window.requestAnimationFrame(() => {
+			const task = app.layoutTasks.shift();
+			if (task) {
+				task.call(this);
+				app.scheduleLayouting();
+			}
+		});
+	}
+};
+
+app.appendLayoutingTask = function (task) {
+	app.layoutTasks.push(task);
+};
+
+app.scheduleLayouting = function () {
+	setTimeout(app._flushLayoutingQueue, 10);
```

Comment:
See above - it seems like the benefit of grouping DOM changes like this is partially lost if we don't actually flush the queue here.

Reply:
omg, I forgot to remember the timer and cancel old one :)